pypeit/core/wavecal/waveio.py

```diff
@@ -441,74 +285,17 @@ def load_unknown_list(lines, unknwn_file=None, all=False):
     line_dict = defs.lines()
     # Load
     if unknwn_file is None:
-        unknwn_file = os.path.join(data.Paths.linelist, 'UNKNWNs.dat')
+        unknwn_file = data.get_linelist_filepath('UNKNWNs.dat')
     line_list = load_line_list(unknwn_file)
     # Cut on input lamps?
     if all:
         return line_list
-    else:
-        msk = np.array([False]*len(line_list))
-        for line in lines:
-            line_flag = line_dict[line]
-            match = line_list['line_flag'] % (2*line_flag) >= line_flag
-            msk[match] = True
-        # Finish
-        return line_list[msk]
-
-
-#def load_spectrum(spec_file, index=0):
-#    """
-#    Load a simple spectrum from input file
-#
-#    Parameters
-#    ----------
-#    spec_file : str
-#        Possible formats are:
-#
-#            - .fits --  Assumes simple ndarray in 0 extension
-#            - .ascii -- Assumes Table.read(format='ascii') will work with single column
-#
-#    Returns
-#    -------
-#
-#    """
-#    import h5py
-#    iext = spec_file.rfind('.')
-#    if 'ascii' in spec_file[iext:]:
-#        tbl = Table.read(spec_file, format='ascii')
-#        key = tbl.keys()[0]
-#        spec = tbl[key].data
-#    elif 'fits' in spec_file[iext:]:
-#        spec = fits.open(spec_file)[0].data
-#    elif 'hdf5' in spec_file[iext:]:
-#        hdf = h5py.File(spec_file, 'r')
-#        if 'arcs' in hdf.keys():
-#            print("Taking arc={:d} in this file".format(index))
-#            spec = hdf['arcs/'+str(index)+'/spec'].value
-#        else:
-#            raise IOError("Not ready for this hdf5 file")
-#    elif 'json' in spec_file[iext:]:
-#        jdict = linetools.utils.loadjson(spec_file)
-#        try:
-#            spec = np.array(jdict['spec'])
-#        except KeyError:
-#            raise IOError("spec not in your JSON dict")
-#    # Return
-#    return spec
-
-
-def write_line_list(tbl, outfile, overwrite=True):
-    """
-    Parameters
-    ----------
-    tbl
-    outfile
-    overwrite (optional), default=True
-    """
-    # Format
-    tbl['wave'].format = '10.4f'
-    # Write
-    with open(outfile,'w') as f:
-        f.write('# Creation Date: {:s}\n'.format(str(datetime.date.today().strftime('%Y-%m-%d'))))
-        tbl.write(f, format='ascii.fixed_width', overwrite=overwrite)
 
+    # Otherwise
+    msk = np.array([False]*len(line_list))
```

Comment:
FWIW, you can also do this like so:
```
msk = np.zeros(len(line_list), dtype=bool)
```

Reply:
Thanks!  I like the `np.zeros` version better.